packages/components/src/components/date-picker/DatePicker.tsx

```diff
@@ -484,6 +490,16 @@ class DatePicker extends Component<
     }
   }
 
+  private handleOnEscAction(e) {
+    const { showPicker } = this.state;
+    if (showPicker) {
+      this.handleOnClose();
+    } else {
+      const element = (e.target as HTMLElement).parentNode;
+      element?.dispatchEvent(new KeyboardEvent(e.type, e));
+    }
+  }
+
```

Comment:
@NamNguyenFPT my point was that the fix/workaround in Lite should be removed as not needed at all, and the only fix to be done (that would fix the Lite issue we had 3 months ago) is my suggestion above.

when we hit ESC in the datepicker, and it is opened, we should close it and stop the propagation of the event, as it is already consumed by the datepicker, and we don't want any parent node to consume that event as it will bubble up the dom tree.

```suggestion
  private handleOnEscAction(e) {
    const { showPicker } = this.state;
    if (showPicker) {
      this.handleOnClose();
      e.stopPropagation();
    }
  }

```